### PR TITLE
feat(history-service): tag Cassandra spans with room_id

### PR DESCRIPTION
**Pull request by hmchangw**
The o11y gocql observer emits a CLIENT span per query but exposes no
per-query attribute hook, so the room a read/write served never reaches
the trace — room_id is only a query argument. Wrap each room-scoped
cassrepo operation in a span carrying room_id (mirroring pkg/atrest's
convention) so the nested query spans inherit it, making per-room
history debugging possible.

Covers the room-scoped reads (message bucket walks, pinned reads) and
writes (pin/unpin, reactions, edit, soft-delete). Methods keyed only by
message_id or thread_room_id are left untouched — no room_id in hand.

Co-Authored-By: Claude Opus 4.8 <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01FQEQfhp7ETVk8t4oB6FD2T

## Summary by CodeRabbit

* **New Features**
  * Added distributed tracing across message storage and history operations.
  * Traces now include relevant room and message context, supporting improved request visibility and operation tracking.
  * Added parent-child span propagation for clearer end-to-end performance analysis.

* **Tests**
  * Added coverage validating trace creation, attributes, context propagation, and nested spans.

### CODE REVIEW

**Comment by julianshen** [comment]
**No actionable comments.** ✅

### Local review memory

_Existing unresolved review discussion is retained here and is not reposted as new findings._

- `message-worker/roomspan_test.go:78` — @coderabbitai: _📐 Maintainability & Code Quality_ _🟡 Minor_ _⚡ Quick win_ **Cover all empty-identifier cases.** Both helpers omit room_id and message_id independently. Add a test for empty messageID in both suites. Add an empty roomID test… ([thread](https://github.com/hmchangw/newchat/pull/190#discussion_r3711555958))

<details>
<summary>📒 Files reviewed (14)</summary>

* `bot-message-worker/roomspan.go`
* `bot-message-worker/roomspan_test.go`
* `bot-message-worker/store_cassandra.go`
* `history-service/internal/cassrepo/messages_by_id.go`
* `history-service/internal/cassrepo/messages_by_room.go`
* `history-service/internal/cassrepo/pin.go`
* `history-service/internal/cassrepo/reactions.go`
* `history-service/internal/cassrepo/repository.go`
* `history-service/internal/cassrepo/roomspan.go`
* `history-service/internal/cassrepo/roomspan_test.go`
* `history-service/internal/cassrepo/write.go`
* `message-worker/roomspan.go`
* `message-worker/roomspan_test.go`
* `message-worker/store_cassandra.go`

</details>
<details>
<summary>⚙️ Rules run (9)</summary>

* `cassandra`
* `distributed-system`
* `elasticsearch`
* `mongodb`
* `nats`
* `oom`
* `performance-and-concurrency`
* `silent-failure`
* `tgd-review`

</details>

File: message-worker/store_cassandra.go [hunk 1]
```diff
@@ -77,6 +80,9 @@ func NewCassandraStore(session *gocql.Session, bucket msgbucket.Sizer, cipher at
 // the legacy plaintext columns are left null. When s.cipher is nil the
 // legacy plaintext batch runs unchanged.
 func (s *CassandraStore) SaveMessage(ctx context.Context, msg *model.Message, sender *cassParticipant, siteID string) error {
+	ctx, span := s.startSpan(ctx, "cassandra.SaveMessage", msg.RoomID, msg.ID)
```
Comment (julianshen) [inline message-worker/store_cassandra.go:83]: _🏷️ observability_ | _🟠 Warning_ | _`distributed-system`_

**Operation spans do not record failed or retried writes**

Every new Cassandra operation span is only ended, never given an error status. A concrete path is an encrypted SaveMessage whose cipher/KMS call fails before any gocql query span exists, or a Cassandra batch that times out after an ambiguous partial apply: the method returns an error and the worker retries/NAKs, but this room/message span remains Unset with no recorded error, making it impossible to distinguish the failed attempt from a completed one using the newly added operation telemetry. Apply the same fix to all newly instrumented methods: capture the returned error, call span.RecordError(err) and span.SetStatus(codes.Error, ...) when non-nil, and optionally mark successful completion codes.Ok.

<details>
<summary>🤖 Prompt for AI Agents</summary>

```
In `message-worker/store_cassandra.go` around line 83: Every new Cassandra operation span is only ended, never given an error status. A concrete path is an encrypted SaveMessage whose cipher/KMS call fails before any gocql query span exists, or a Cassandra batch that times out after an ambiguous partial apply: the method returns an error and the worker retries/NAKs, but this room/message span remains Unset with no recorded error, making it impossible to distinguish the failed attempt from a completed one using the newly added operation telemetry. Apply the same fix to all newly instrumented methods: capture the returned error, call span.RecordError(err) and span.SetStatus(codes.Error, ...) when non-nil, and optionally mark successful completion codes.Ok.

Fix only this issue, keep the change minimal, and make sure the tests still pass.
```

</details>

Comment (julianshen) [inline message-worker/store_cassandra.go:83]: [inline permission test] tGDBot inline-comment permission test for PR #190.

**Comment by hmchangw** [comment]
Implemented in another PR.
